test/unit/render/glyph_manager.test.js

```diff
@@ -60,3 +60,41 @@ test('GlyphManager generates CJK PBF locally', (t) => {
         t.end();
     });
 });
+
+test('GlyphManager generates Katakana PBF locally', (t) => {
+    t.stub(GlyphManager, 'TinySDF').value(class {
+        // Return empty 30x30 bitmap (24 fontsize + 3 * 2 buffer)
+        draw() {
+            return new Uint8ClampedArray(900);
+        }
+    });
+
+    const manager = new GlyphManager((url) => ({url}), 'sans-serif');
+    manager.setURL('https://localhost/fonts/v1/{fontstack}/{range}.pbf');
+
+    // Katakana letter te
+    manager.getGlyphs({'Arial Unicode MS': [0x30c6]}, (err, glyphs) => {
+        t.ifError(err);
+        t.equal(glyphs['Arial Unicode MS'][0x30c6].metrics.advance, 24);
+        t.end();
+    });
+});
+
+test('GlyphManager generates Hiragan PBF locally', (t) => {
+    t.stub(GlyphManager, 'TinySDF').value(class {
+        // Return empty 30x30 bitmap (24 fontsize + 3 * 2 buffer)
+        draw() {
+            return new Uint8ClampedArray(900);
+        }
+    });
+
+    const manager = new GlyphManager((url) => ({url}), 'sans-serif');
+    manager.setURL('https://localhost/fonts/v1/{fontstack}/{range}.pbf');
+
+    //Hiragan letter te
+    manager.getGlyphs({'Arial Unicode MS': [0x3066]}, (err, glyphs) => {
+        t.ifError(err);
+        t.equal(glyphs['Arial Unicode MS'][0x3066].metrics.advance, 24);
```

Comment:
I'm not sure I fully understand what the tests do. I get that you're stubbing the `draw` method of the static `TinySDF` object. But how does this test ensure that `TinySDF.draw` was called as opposed to going through the regular pathway? In either case, wouldn't we expect the glyph to be 24px wide?

Reply:
`GlyphManager#loadGlyphRange` would be the other code path, and that is not stubbed, so the test will error out if it tries to go down that path. The `metrics` test is to ensure that the code path called `_tinySDF` successfully and returned a `StyleGlyph` successfully.

Reply:
Ah ok, that makes sense.